idaes/core/scaling/custom_scaler_base.py

```diff
@@ -0,0 +1,608 @@
+import idaes.logger as idaeslog
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+CSCONFIG = CONFIG()
+
+DEFAULT_UNIT_SCALING = {
+    # "QuantityName: (reference units, scaling factor)
+    "Temperature": (units.K, 1e-2),
+    "Pressure": (units.Pa, 1e-5),
+}
+
+
+class CustomScalerBase(ScalerBase):
+    """
+    Base class for custom scaling routines.
+
+    """
+
+    CONFIG = ScalerBase.CONFIG()
+
+    # Common data structures for default scaling
+    # DEFAULT_SCALING_FACTORS = {"component_local_name": DEFAULT_SCALING}
+    DEFAULT_SCALING_FACTORS = None
+
+    # UNIT_SCALING_FACTORS = {"units": UNIT_BASED_SCALING}
+    UNIT_SCALING_FACTORS = copy(DEFAULT_UNIT_SCALING)
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+        if self.DEFAULT_SCALING_FACTORS is not None:
+            self.default_scaling_factors = copy(self.DEFAULT_SCALING_FACTORS)
+        else:
+            self.default_scaling_factors = {}
+
+        if self.UNIT_SCALING_FACTORS is not None:
+            self.unit_scaling_factors = copy(self.UNIT_SCALING_FACTORS)
+        else:
+            self.unit_scaling_factors = {}
+
+    def scale_model(
+        self,
+        model,
+        first_stage_fill_in: list = None,
+        second_stage_fill_in: list = None,
+        submodel_scalers: dict = None,
+    ):
+        """
+        Default model scaling routine.
+
+        This method performs a four-step scaling routine:
+
+        1. Scale variables using variable_scaling_routine
+        2. Perform first-stage scaling factor fill in using user provided method(s), called in order declared
+        3. Scale constraints using constraint_scaling_routine
+        4. Perform second-stage scaling factor fill in using user provided method(s), called in order declared
+
+        Args:
+            model: model to be scaled
+            first_stage_fill_in: list of methods to use for first-stage scaling factor fill in
+            second_stage_fill_in: list of methods to use for second-stage scaling factor fill in
+            submodel_scalers: dict of Scalers to use for sub-models, keyed by submodel local name
+
+        Returns:
+            None
+        """
+        # Step 1: Call variable scaling routine
+        self.variable_scaling_routine(
+            model, overwrite=self.config.overwrite, submodel_scalers=submodel_scalers
+        )
+        # Step 2: Call variable fill in
+        if first_stage_fill_in is not None:
+            for i in first_stage_fill_in:
+                i(model)
+
+        # Step 3: Call constraint scaling routine
+        self.constraint_scaling_routine(
+            model, overwrite=self.config.overwrite, submodel_scalers=submodel_scalers
+        )
+
+        # Step 4: Call constraint fill in
+        if second_stage_fill_in is not None:
+            for i in second_stage_fill_in:
+                i(model)
+
+        # TODO: Consider how to make this work
```

Comment:
@dallan-keylogic and I think we have a way to avoid the need for this which he is working on. I'll remove the TODO and let him decide if he wants to create an issue or not.